### ISSUE TITLE
Lack of support for modern ECC

### ISSUE THREAD

**Issue by MichalStaruch** [w3c/webauthn#1124]
I'd like to know what's WG opinion about cryptography concerns related to WebAuthn, which were raised in [Paragon Initiative analysis](https://paragonie.com/blog/2018/08/security-concerns-surrounding-webauthn-don-t-implement-ecdaa-yet), especially those related to WebAuthn vs modern ECC:
- no support for [EdDSA](https://tools.ietf.org/html/rfc8032) - which is considered superior to ECDSA, and already supported in [TLSv1.3](https://tools.ietf.org/html/rfc8446),
- no support for [Curve25519 and Curve448](https://tools.ietf.org/html/rfc7748) - both are considered superior to NIST curves, and are supported in [TLSv1.3](https://tools.ietf.org/html/rfc8446), too. It's also worth mentioning that Curve25519 is already implemented in every modern browser.

When developing new standard like WebAuthn related to security of wider Internet audience - one that might help shaping web browsers - it would reasonable to take into account work already done by IETF WGs.

**Comment by herrjemand** [comment]
Can be closed

**Comment by herrjemand** [comment]
The curves been defined in COSE RFC https://tools.ietf.org/html/rfc8152
And in FIDO Registry: https://fidoalliance.org/specs/fido-v2.0-rd-20180702/fido-server-v2.0-rd-20180702.html

**Comment by arianvp** [comment]
I don't think this is implemented yet.

The `signature` field in websauthn has nothing to do with COSE. And doesn't say it supports  EdDSA in the standard. So it's underspecified what to do when a fido2 device provides an edDSA signature.

**Comment by bdewater** [comment]
@arianvp I believe the changes in https://github.com/w3c/webauthn/pull/1420 cover this?

**Comment by arianvp** [comment]
Not as far as I can see in that PR. The signature field is still not
specified for each algorithm. Only for ECDSA and RSA.

But let me read the latest draft of the standard before I make these
claims. Maybe it is already mentioned in another part.

I'll open an issue if it's the case after I read the latest draft.

**Comment by dwaite** [comment]
I believe the signature value is meant to be the same as the COSE "signature" field, which is a bstr.

**Comment by arianvp** [comment]
That's not true.  The signature field can be any amount of formats, for backwards compatibility reasons. For `ES256`, `RS256` and `PS256` it does *not* correspond to the signature format of COSE. https://www.w3.org/TR/webauthn/#signature-attestation-types

Result is the `signature` field in assertions is only defined for `ES256` `RS256` and `PS256` if I read precisely but is undefined for any other COSEAlgorithmIdentifiers 

The **NOTE** does mention that it is "recommended" that any new signature formats will directly correspond to the COSE signature field, but the NOTE is not normative

Hence; the `signature` field seems underspecified to me currently and it's not clear to me as an implementor of a Relying Party how it should be interpreted from the standard alone.

**Comment by arianvp** [comment]
The lastest draft has the same issue https://w3c.github.io/webauthn/#sctn-signature-attestation-types

It does not specify what the _format_ is for `signature`   when it is not one of `ES256, RS256, PS256`

**Comment by equalsJeffH** [comment]
thanks @arianvp -- I agree that the language in https://w3c.github.io/webauthn/#sctn-signature-attestation-types is problematic in various ways and underspecified.  submitted new issue #1441 to cover this.

**Comment by selfissued** [comment]
Note that 25519 has been in the IANA Cose Algorithms Registry https://www.iana.org/assignments/cose/cose.xhtml since July 2017, since it was defined by RFC 8152.
